rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/api/ExamTermApi.java

```diff
@@ -0,0 +1,79 @@
+package org.niewidoczniakademicy.rezerwacje.api;
+
+import lombok.AllArgsConstructor;
+import org.niewidoczniakademicy.rezerwacje.core.model.database.CourseOfStudy;
+import org.niewidoczniakademicy.rezerwacje.core.model.database.ExamTerm;
+import org.niewidoczniakademicy.rezerwacje.core.model.database.Room;
+import org.niewidoczniakademicy.rezerwacje.repository.CourseOfStudyRepository;
+import org.niewidoczniakademicy.rezerwacje.repository.ExamTermRepository;
+import org.niewidoczniakademicy.rezerwacje.repository.RoomRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Set;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@RestController
+@RequestMapping(path = "exam-terms")
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public class ExamTermApi {
+
+    private final ExamTermRepository examTermRepository;
+    private final CourseOfStudyRepository cosRepository;
+    private final RoomRepository roomRepository;
+
+    @GetMapping
+    public List<ExamTerm> getAll() {
+        return examTermRepository.findAll();
+    }
+
+    @GetMapping(path = "{id}")
+    public ExamTerm getOne(@PathVariable Long id) {
+        return examTermRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, "Unable to find resource " + id));
+    }
+
+    @GetMapping(path = "room/{id}")
+    public Set<ExamTerm> getByRoom(@PathVariable Long id) {
+        return roomRepository.findById(id).map(Room::getExamTerms)
+                .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, "Unable to find resource " + id));
+    }
+
+    @GetMapping(path = "cos/{id}")
+    public Set<ExamTerm> getByExam(@PathVariable Long id) {
+        return cosRepository.findById(id).map(CourseOfStudy::getExamTerms)
+                .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, "Unable to find resource " + id));
+    }
+
+    @GetMapping(path = "cos/{cosId}/room/{roomId}")
+    public ExamTerm getOneByRoomAndExam(@PathVariable Long roomId,
+                                        @PathVariable Long cosId) {
+        return examTermRepository.findByRoomIdAndCourseOfStudyId(roomId, cosId)
+                .orElseThrow(() -> new ResponseStatusException(NOT_FOUND,
+                        "Unable to find resource with room id " + roomId + " and course of study id" + cosId));
+    }
+
+    @PostMapping
+    @ResponseStatus(value = HttpStatus.CREATED)
+    public void addExamTerm(@RequestParam @DateTimeFormat(pattern = "dd-MM-yyyy") LocalDate day, // TODO Global config
```

Comment:
W ogóle to nie lepiej w jakimś body wysyłać?

Reply:
Pewnie lepiej, ale to trzeba by było tak jak napisałeś wcześniej osobny model dla bazy i osobny dla resta i parsować między tym. A tutaj jest problem z tą datą, nie udało mi się tego globalnie dla projektu ustawić tak jak tutaj [https://www.baeldung.com/spring-date-parameters#convert-date-parameters-at-the-application-level](url). Jakby ktoś to dodał to poprawie. Tym bardziej, że trzeba zrobić validacje poprawnie przesyłanej daty, a to pewnie dobrze by było faktycznie w restowym, a nie bazodanowym modelu trzymać. Dodam to, ale nie w tym issue.